opentelemetry-api/src/opentelemetry/attributes/__init__.py

```diff
@@ -173,19 +278,24 @@ def __setitem__(self, key: str, value: types.AttributeValue) -> None:
                 self.dropped += 1
                 return
 
-            value = _clean_attribute(key, value, self.max_value_len)  # type: ignore
-            if value is not None:
-                if key in self._dict:
-                    del self._dict[key]
-                elif (
-                    self.maxlen is not None and len(self._dict) == self.maxlen
-                ):
-                    if not isinstance(self._dict, OrderedDict):
-                        self._dict = OrderedDict(self._dict)
-                    self._dict.popitem(last=False)  # type: ignore
-                    self.dropped += 1
-
-                self._dict[key] = value  # type: ignore
+            if self._extended_attributes:
+                value = _clean_extended_attribute(
```

Comment:
Do we already know that `key` is a string a this point -- in which case we can call `_clean_extended_attribute_value` instead? If so, maybe `_clean_extended_attribute` can go away since this is the only use of that function.

Reply:
We don't, the key is validated inside `_clean_extended_attribute`